### PR TITLE
Added new reporter: MARKDOWN_SUMMARY Reporter

### PR DESCRIPTION
**Pull request by saishivarcr**
New reporter **MARKDOWN_SUMMARY_REPORTER**, allows saving MegaLinter results summary as a markdown file. This file can be further utilised to add comments on the pull request (PR) from Jenkins and other continuous integration (CI) tools.

### CODE REVIEW

**Comment by echoix** [comment]
Do you use or intend to use it with Jenkins? I think you could expand on the use cases in your documentation for better discovery.

I see another useful use-case with this: something that implements https://github.com/oxsecurity/megalinter/issues/1489

I once tried to start it, but only finished the research phase. 
https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary

The only missing change to enable this functionality would be to check if the `GITHUB_STEP_SUMMARY` env var exists, and if so, save the Markdown file there.

**Comment by nvuillam** [comment]
@saishivarcr thanks a lot for your PR :)

we already have reporters for Github, Gitlab & Azure (and probably bitbucket someday) -> https://megalinter.io/latest/reporters/

If it is for jenkins purpose, why not some JENKINS_COMMENT_REPORTER, that would handle the overwrite on existing comment etc... like the other CI platforms reporters ? ( or even a standalone Jenkins Plugin ? 👼 )

Otherwise, I'm ok to merge your PR, but with the following pre-requisites:
- rename it as MARKDOWN_SUMMARY_REPORTER
- deactivate it by default

**Comment by saishivarcr** [comment]
Thanks @echoix for providing your feedback.

> Do you use or intend to use it with Jenkins? I think you could expand on the use cases in your documentation for better discovery.

Yes, my primary use case is to post the job summary as a PR comment from the Jenkins pipeline job. Once the summary report is available in markdown format, one can write a custom script to post it to Bitbucket or other repository hosting platforms (non-Github/Gitlab) without waiting for such functionality to be made officially available.

> I see another useful use-case with this: something that implements #1489
> 
> I once tried to start it, but only finished the research phase. https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/ https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary
>
> The only missing change to enable this functionality would be to check if the `GITHUB_STEP_SUMMARY` env var exists, and if so, save the Markdown file there.

Yes, this use case can now be covered easily with minimal enhancements. I will work on this in a separate PR.

**Comment by saishivarcr** [comment]
Thanks @nvuillam for your feedback.

> If it is for jenkins purpose, why not some JENKINS_COMMENT_REPORTER, that would handle the overwrite on existing comment etc... like the other CI platforms reporters ? ( or even a standalone Jenkins Plugin ? 👼 )
> 

I wanted to make it generic and not tied to either Jenkins or other CI platforms. The Jenkins plugin is a great idea, but I don't have much experience in developing one. This can be considered for further enhancements in upcoming versions.

> * rename it as MARKDOWN_SUMMARY_REPORTER

Yes, this is a good suggestion; I will make this change.

> * deactivate it by default, except if you find a way to post it by default as Job Summary like suggested by @echoix  :)

By default, this feature is deactivated. Could you kindly verify this once again through a code review on your end?

**Comment by nvuillam** [comment]
@saishivarcr I already checked your code and it seems right to me, so once the renaming done, I'll merge then it'll be shipped in the next release 🥳 🎈

File: megalinter/constants.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 DEFAULT_DOCKER_WORKSPACE_DIR = "/tmp/lint"
 DEFAULT_REPORT_FOLDER_NAME = "megalinter-reports"
 DEFAULT_SARIF_REPORT_FILE_NAME = "megalinter-report.sarif"
+DEFAULT_SUMMARY_REPORT_FILE_NAME = "megalinter-summary.md"
```
Comment (nvuillam) [inline megalinter/constants.py:22]: He plz rename into megalinter-report.md :)

File: docs/assets/images/MarkdownSummaryReporter_1.png
Comment (nvuillam) [inline docs/assets/images/MarkdownSummaryReporter_1.png:1]: @saishivarcr plz can you also add a screenshot with the result of the markdown ? ( CTRL + Shift + V in VsCode :) )

File: docs/reporters/MarkdownSummaryReporter.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+title: Summary Reporter for MegaLinter
+description: Generates a summary of SAST results in Markdown within a file named 'megalinter-report.md', located in the report folder.
+---
+
+# Summary Reporter
```
Comment (nvuillam) [inline docs/reporters/MarkdownSummaryReporter.md:6]: "Markdown Summary Reporter" plz :)

**Review by nvuillam — CHANGES_REQUESTED** [review]
Great, many thanks :)
Plz check comments for tiny updates  :)

File: docs/assets/images/MarkdownSummaryReporter_1.png
Comment (saishivarcr) [reply docs/assets/images/MarkdownSummaryReporter_1.png:1]: This is added here: docs/assets/images/MarkdownSummaryReporter_2.png

File: docs/reporters/MarkdownSummaryReporter.md (hunk 2 above)
Comment (saishivarcr) [reply docs/reporters/MarkdownSummaryReporter.md:6]: Done.

**Review by nvuillam — APPROVED** [review]
Great :)

**Pull request merged**